waku/v2/protocol/waku_rln_relay/rln_relay.nim

```diff
@@ -0,0 +1,421 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/[algorithm, sequtils, strutils, tables, times, os, deques],
+  chronicles, options, chronos, stint,
+  confutils,
+  strutils,
+  web3, json,
+  web3/ethtypes,
+  eth/keys,
+  libp2p/protocols/pubsub/rpc/messages,
+  libp2p/protocols/pubsub/pubsub,
+  stew/results,
+  stew/[byteutils, arrayops]
+import
+  ./group_manager,
+  ./rln,
+  ./conversion_utils,
+  ./constants,
+  ./protocol_types,
+  ./protocol_metrics
+import
+  ../../utils/time,
+  ../waku_keystore
+
+logScope:
+  topics = "waku rln_relay"
+
+type WakuRlnConfig* = object
+  rlnRelayDynamic*: bool
+  rlnRelayPubsubTopic*: PubsubTopic
+  rlnRelayContentTopic*: ContentTopic
+  rlnRelayMembershipIndex*: Option[uint]
+  rlnRelayEthContractAddress*: string
+  rlnRelayEthClientAddress*: string
+  rlnRelayEthAccountPrivateKey*: string
+  rlnRelayEthAccountAddress*: string
+  rlnRelayCredPath*: string
+  rlnRelayCredentialsPassword*: string
+
+proc createMembershipList*(n: int): RlnRelayResult[(
+    seq[(string, string, string, string)], string
+  )] =
```

Comment:
When you have these types `(string, string, string, string)` my recommendation is that you define a Type name/alias. For example:

`type MembershipList = (string, string, string, string)`

Reply:
Addressed in 09da0a85, thanks for the review!